ecosystem/platform/server/app/controllers/wallets_controller.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Copyright (c) Aptos
+# SPDX-License-Identifier: Apache-2.0
+
+class WalletsController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    wallet_params = params.require(:wallet).permit(
+      :network, :wallet_name, :public_key
+    )
+
+    challenge = params.require(:challenge)
+    return head :forbidden unless challenge.match(/[0-9]{24}/)
```

Comment:
the challenge is going to be a bit diff, should take a look at #wallet to see the new format